### PR TITLE
cbprintf: small improvements to the unit test

### PR DESCRIPTION
**Pull request by npitre**

### CODE REVIEW

File: tests/unit/cbprintf/main.c [hunk 1]
```diff
@@ -224,9 +230,11 @@ static inline bool prf_check(const char *expected,
 	};
 
 	const char *str = buf;
-	const char *sp = str;
-	int rc = match_pfx(&str);
+	const char *sp;
+	int rc;
 
+	sp = str;
```
Comment (pabigot) [inline tests/unit/cbprintf/main.c:236]: I don't see a huge value in leaving the values undefined at the definition, then assigning them immediately after.  But personal style, I guess.  I might automatically change them back if I have to edit this file again.

File: tests/unit/cbprintf/main.c [hunk 2]
```diff
@@ -247,10 +254,8 @@ static inline bool prf_check(const char *expected,
 	rc = (*str != 0);
 	if (rc != 0) {
 		return prf_failed(&ctx, str, "no eos %02x\n", *str);
-		return false;
```
Comment (pabigot) [inline tests/unit/cbprintf/main.c:250]: Yeah; obviously I didn't look closely at this.

**Review by pabigot — APPROVED** [review]

**Comment by npitre** [comment]
> I don't see a huge value in leaving the values undefined at the definition, then assigning them immediately after.  But personal style, I guess.

This function has several code blocks following this pattern:

```
	sp = str;
	rc = match_...
```

Except for the first one which didn't follow that pattern. I just made it so.
It's slightlyeasier to review when presented uniformly. And as you know, the compiler doesn't care.

> I might automatically change them back if I have to edit this file again.

If you wish.

**Comment by pabigot** [comment]
> > I might automatically change them back if I have to edit this file again.
> If you wish.

That was just an observation; I very much prefer to see variables initialized when defined, so I may not remember that there's an defensible justification for not doing it.  If we didn't have the "blank line between declarations and code" rule I'd just treat the definitions as code and it wouldn't look quite so odd.

**Review by nashif — APPROVED** [review]

**Pull request merged**